tensorpack/dataflow/format.py

```diff
@@ -103,6 +118,8 @@ def get_data(self):
                 if k != '__keys__':
                     yield [k, v]
         else:
+            if not hasattr(self, "rng"):
+                self.reset_state()
```

Comment:
The trainer and TestDataSpeed do that for you. But often you are supposed to call that yourself, although it doesn't always give you exception.